src/internal.rs

```diff
@@ -462,18 +464,43 @@ impl Pal {
         self.data.is_empty()
     }
 
-    pub fn as_slice<BD: BitDepth>(&self) -> &[[[BD::Pixel; 8]; 3]] {
-        as_chunks::<3, [BD::Pixel; 8]>(
-            as_chunks::<8, BD::Pixel>(BD::cast_pixel_slice(&self.data)).0,
-        )
-        .0
+    pub fn index<'a: 'b, 'b, BD: BitDepth>(
+        &'a self,
+        index: usize,
+    ) -> DisjointImmutGuard<'b, AlignedVec64<u8>, [[BD::Pixel; 8]; 3]> {
+        let guard: DisjointImmutGuard<'a, AlignedVec64<u8>, [u8]> = self.data.index(
+            index * 8 * 3 * mem::size_of::<BD::Pixel>()
+                ..(index + 1) * 8 * 3 * mem::size_of::<BD::Pixel>(),
+        );
+        guard.cast()
```

Comment:
```suggestion
        type E<BD: BitDepth> = [[BD::Pixel; 8]; 3];
        
        let guard: DisjointImmutGuard<'a, AlignedVec64<u8>, [u8]> = self.data.index(
            index * mem::size_of::<E<BD>>()
                ..(index + 1) * mem::size_of::<E<BD>>(),
        );
        guard.cast::<E<BD>>()
```

I'd like to avoid the possibility of arithmetic errors turning into panics or UB.  This way they'd be a type error.  Given there's `unsafe`ty involved, I think the extra precaution is worth it.  The same can be done for `index_mut`.  We also shouldn't be manually computing array sizes with arithmetic.  Size calculation of arrays is a bit more complex (e.x. if alignment every becomes an issue, it changes); we should leave it up to the compiler.